src/platform/plugins/shared/unified_search/public/search_bar/create_search_bar.tsx

```diff
@@ -253,6 +253,7 @@ export function createSearchBar({
             }
             savedQuery={savedQuery}
             onQuerySubmit={defaultOnQuerySubmit(props, data.query, query)}
+            onQueryChange={props.onQueryChange}
```

Comment:
During my investigation I figured to, which onQueryChange is in the searcher props, we don't use it under the hood. seems we have no consumer for this, but we might need it in the future

Reply:
I'm always a little skeptical about changing such brittle and widely consumed code... I also don't get any results when checking for references, but that seemed to be the case for all props in `SearchBarOwnProps` that I checked, so something odd is going on. Just not sure if we should be worried about any odd side effects here since I'm unfamiliar with the code.

Reply:
We don't have any more context at the moment, as we haven't had the bandwidth to really take ownership of this plugin quite yet.

Reply:
I've removed it for now, let's add it when we need it 👍